marconi-mamba/src/Marconi/Bootstrap.hs

```diff
@@ -51,8 +53,10 @@ bootstrapUtxoIndexers
     -> IO ()
 bootstrapUtxoIndexers (CliArgs socket dbPath _ networkId targetAddresses) env =
     do
-        let qsem = env ^. queryEnv . queryTMVar
-            indexers = combineIndexers [( queryAwareUtxoWorker qsem targetAddresses , dbPath)]
+        let (UtxoQueryTMVar qTMVar) = env ^. queryEnv . queryTMVar
+            callbackIndexer :: QApi.UtxoIndex -> IO QApi.UtxoIndex
+            callbackIndexer index = atomically $ (QApi.writeTMVar qTMVar  index)  >>  pure index
+            indexers = combineIndexers [( utxoWorker  callbackIndexer (Just targetAddresses) , dbPath)]
```

Comment:
The utxo worker uses it to save the refreshed state of the VSplit storage after inserts/rollback. The VSplit [Storage](https://github.com/input-output-hk/plutus-apps/blob/main/rewindable-index/src/RewindableIndex/Index/VSplit.hs#L33) pointers, cursr, esize, ... constantly change which will effect the query. And this callback stuff is only needed in the context of mamba. and hence the need for this callback. Do you know of a different way to get access to the Storage?

Reply:
I would probably inline it into the utxo worker, but this also works.

Reply:
Yes, I can do that with the next PR, if we still need it. 
The main reason, may be not a good one, was that I thought it was more clear as to what the call back is doing, readability,  rather than in lining it.